airflow/opt/providers/etna/etna/auth/etna_cookie_auth.py

```diff
@@ -107,11 +109,59 @@ def sync_project_role(self, project_name: str) -> Role:
         cond = DagModel.tags.any(DagTag.name == project_name)
         dags_query = dags_query.filter(cond)
 
+        project_permissions = [
+            # From standard Viewer
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_AUDIT_LOG),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_DEPENDENCIES),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_CODE),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG_RUN),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_IMPORT_ERROR),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_JOB),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_INSTANCE),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_LOG),
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_WEBSITE),
+            (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_BROWSE_MENU),
+            (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_DAG_DEPENDENCIES),
+            (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_DAG_RUN),
+            (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_JOB),
+            (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_AUDIT_LOG),
+            (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_TASK_INSTANCE),
+            (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_DOCS),
+
+            # From standard User
+            (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG),
+            (permissions.ACTION_CAN_CREATE, permissions.RESOURCE_TASK_INSTANCE),
+            (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_TASK_INSTANCE),
+            (permissions.ACTION_CAN_CREATE, permissions.RESOURCE_DAG_RUN),
+            (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_DAG_RUN),
+
+            # From standard Operator
+            (permissions.ACTION_CAN_ACCESS_MENU, permissions.RESOURCE_ADMIN_MENU),
+
+            # For plugins
+            (permissions.ACTION_CAN_ACCESS_MENU, ETNA_DOCS_MENU_TITLE),
+            (permissions.ACTION_CAN_ACCESS_MENU, CODE_EDITOR_MENU_LABEL),
+        ]
+
         for dag in dags_query.all():
-            perm_view = self.add_permission_view_menu(
+            project_permissions.append((
                 permissions.ACTION_CAN_READ,
                 resource_name_for_dag(dag.dag_id),
-            )
-            self.add_permission_role(role, perm_view)
+            ))
+
+        # Convert to Permissions
+        project_permissions = [self.add_permission_view_menu(*p) for p in project_permissions]
+
+        # Remove any old permissions on the role
+        for role_perm in role.permissions:
```

Comment:
Cool.  Originally I knew i had left out this 'removal' part of syncing, but it's wise to add it.